ironfish/src/network/peers/peerManager.ts

```diff
@@ -609,37 +617,37 @@ export class PeerManager {
   /** For a given peer, try to find a peer that's connected to that peer
    * including itself to broker a WebRTC connection to it
    * */
-  private getBrokeringPeer(peer: Peer): Peer | null {
+  private getBrokeringPeers(peer: Peer): Peer[] {
     if (peer.state.type === 'CONNECTED') {
       // Use the existing connection to the peer to broker the connection
-      return peer
+      return [peer]
     }
 
     if (peer.state.identity === null) {
       // Cannot find a brokering peer of an unidentified peer
-      return null
+      return []
     }
 
-    // Find another peer to broker the connection
-    const candidates = []
-
     // The peer candidate map tracks any brokering peer candidates
     const val = this.peerCandidates.get(peer.state.identity)
     if (!val) {
-      return null
+      return []
     }
 
+    // Find another peer to broker the connection
+    const candidates = []
+
     for (const neighbor of val.neighbors) {
       const neighborPeer = this.identifiedPeers.get(neighbor)
-      if (!neighborPeer || neighborPeer.state.type !== 'CONNECTED') {
+
+      if (neighborPeer && neighborPeer.state.type === 'CONNECTED') {
+        candidates.push(neighborPeer)
+      } else {
         val.neighbors.delete(neighbor)
-        continue
       }
-
-      candidates.push(neighborPeer)
     }
 
-    return ArrayUtils.sample(candidates)
+    return candidates
```

Comment:
What do you think about adding the shuffle here, since the callsites (that are actually interested in the results) call shuffle anyway? I'd also suggest adding a `hasBrokeringPeer` function, for the cases where we don't want the overhead of materializing the array. Specifically these spots, really: https://github.com/iron-fish/ironfish/pull/4401/files#diff-52ccbf1a59c642f18891176e72f472cab95d1439d6c9e458741facc0394ac8e7R357
https://github.com/iron-fish/ironfish/pull/4401/files#diff-52ccbf1a59c642f18891176e72f472cab95d1439d6c9e458741facc0394ac8e7R261

Reply:
Moved the shuffling operation into that function. 

I think the problem with adding a hasBrokeringPeer function is that to determine if there are brokering peers, you have to go through the entire logic of the getBrokeringPeers function because the last loop in that function can still return an empty array. 

```
 const candidates = []

    for (const neighbor of val.neighbors) {
      const neighborPeer = this.identifiedPeers.get(neighbor)

      if (neighborPeer && neighborPeer.state.type === 'CONNECTED') {
        candidates.push(neighborPeer)
      } else {
        val.neighbors.delete(neighbor)
      }
    }

    return ArrayUtils.shuffle(candidates)
```

Reply:
It could still be a lot lighter, if I'm not misunderstanding. Something like:
``` typescript
for (const neighbor of val.neighbors) {
  const neighborPeer = this.identifiedPeers.get(neighbor)
  if (neighborPeer && neighborPeer.state.type === 'CONNECTED') {
    return true
  }
}
return false
```

You're still iterating but you're not creating the new array nor doing the shuffle

Reply:
I'm not sure what the unintended consequence of what you have above which eliminates the `val.neighbors.delete(neighbor)` call